packages/mobile/src/components/details-tile/DetailsTilePremiumAccess.tsx

```diff
@@ -23,17 +24,10 @@ export const DetailsTilePremiumAccess = ({
   doesUserHaveAccess,
   style
 }: DetailsTilePremiumAccessProps) => {
-  const { follow_user_id: followUserId, tip_user_id: tipUserId } =
-    premiumConditions ?? {}
-  const users = useSelector((state) =>
-    getUsers(state, {
-      ids: [followUserId, tipUserId].filter(removeNullable)
-    })
-  )
-  const followee = followUserId ? users[followUserId] : null
-  const tippedUser = tipUserId ? users[tipUserId] : null
   const shouldDisplay =
-    !!premiumConditions.nft_collection || followee || tippedUser
+    isPremiumContentCollectibleGated(premiumConditions) ||
```

Comment:
Unfortunately @saliou and I discovered that `nft_collection` specifically can be undefined, bc during upload flow user could set the track to collectible gated without specifying a collection yet. Kind of awkward because at every point afterwards all the fields should be defined.

Reply:
ah good point. Probably good to keep it this way for now tho bc `usdc_purchase` is a valid option but I haven't written the logic to exclude that yet?